### PR TITLE
Update pro encoding documentation to use class based definition

### PR DESCRIPTION
**Pull request by joelzwarrington**
Fixes https://github.com/rmosolgo/graphql-ruby/issues/3991

### CODE REVIEW

File: guides/pro/encoders.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ pro: true
 Encoders can be created with `Encoder.define { ... }`:
```
Comment (rmosolgo) [inline guides/pro/encoders.md:25]:
```suggestion
Encoders can be created with by subclassing `GraphQL::Pro::Encoder`: 
```

File: guides/pro/encoders.md [hunk 2]
```diff
@@ -55,17 +55,16 @@ If you implement your own connections, you can access the encoder's encryption m
 Encrypt IDs by using encoders in `Schema#id_from_object` and `Schema#object_from_id`:
```
Comment (rmosolgo) [inline guides/pro/encoders.md:55]:
```suggestion
Encrypt IDs by using encoders in `Schema.id_from_object` and `Schema.object_from_id`:
```

**Review by rmosolgo — COMMENTED** [review]
Thanks for updating this! I don't know how I overlooked it for so long 🤦 I found a couple other spots that are due for updates.

File: guides/pro/encoders.md (hunk 1 above)
Comment (joelzwarrington) [reply guides/pro/encoders.md:25]: Ahhh, thanks for catching that!

File: guides/pro/encoders.md (hunk 2 above)
Comment (joelzwarrington) [reply guides/pro/encoders.md:55]: I'm blind :eye: :x: thanks!

**Comment by joelzwarrington** [comment]
Good for a re-review! 😄  ✋  5️⃣

File: guides/pro/encoders.md [hunk 3]
```diff
@@ -22,10 +22,10 @@ pro: true
 
 ## Defining an Encoder
 
-Encoders can be created with `Encoder.define { ... }`:
+Encoders can be created with by subclassing `GraphQL::Pro::Encoder`:
```
Comment (joelzwarrington) [inline guides/pro/encoders.md:25]:
```suggestion
Encoders can be created by subclassing `GraphQL::Pro::Encoder`: 
```

File: guides/pro/encoders.md [hunk 4]
```diff
@@ -77,9 +76,17 @@ You can combine several encoders into a single chain of versioned encoders. Pass
...
+class OldSecureEncoder < GraphQL::Pro::Encoder
+  # ...
+end
+
+class LegacyInsecureEncoder < GraphQL::Pro::Encoder
+  # ...
+end
 
 # Then order them by priority:
 VersionedEncoder = GraphQL::Pro::Encoder.versioned(
```
Comment (joelzwarrington) [inline guides/pro/encoders.md:92]: @rmosolgo is this still how versioned encoders are defined?

I don't recall how we started defining this and we've diverged a little bit into our own encoder class

Comment (rmosolgo) [reply guides/pro/encoders.md:92]: Yup, it's the same :+1:

**Comment by rmosolgo** [comment]
Thanks for updating this!

**Pull request merged**